src/celeritas/ext/g4vg/LogicalVolumeConverter.cc

```diff
@@ -75,11 +76,21 @@ auto LogicalVolumeConverter::make_volume_map() const -> MapLvVolId
  */
 auto LogicalVolumeConverter::construct_base(arg_type g4lv) -> result_type
 {
-    vecgeom::VUnplacedVolume const* shape
-        = this->convert_solid_(*g4lv.GetSolid());
-
-    if (dynamic_cast<GenericSolidBase const*>(shape))
+    vecgeom::VUnplacedVolume const* shape = nullptr;
+    try
+    {
+        shape = this->convert_solid_(*g4lv.GetSolid());
+    }
+    catch (celeritas::RuntimeError const& e)
     {
+        CELER_LOG(error) << "Failed to convert solid type '"
+                         << g4lv.GetSolid()->GetEntityType() << "' named '"
+                         << g4lv.GetSolid()->GetName()
+                         << "': " << e.details().what;
+        shape = this->convert_solid_.to_sphere(*g4lv.GetSolid());
+        CELER_LOG(warning)
+            << "Replaced unknown solid with sphere with capacity "
+            << shape->Capacity() << " [cm^3]";
```

Comment:
Spherical cow?

Reply:
A spherical cow is better than a 10x size cow that crashes when you try to put it on a GPU. 🐮